### PR TITLE
FIXED: Implement Dropdown for Hidden Breadcrumb Paths Across the Platform

### LINKED ISSUE #9369: Implement Dropdown for Hidden Breadcrumb Paths Across the Platform
Follow-up : https://github.com/ohcnetwork/care_fe/pull/8860#issuecomment-2428084413

**Current Behavior**

The breadcrumb navigation displays the entire path in a single line, which consumes excessive space and negatively impacts the user experience, especially on screens of varying sizes.

**Expected Behavior**

Display only the current page in the breadcrumb by default. Move the additional path segments (non-current pages) into a dropdown menu to optimize space utilization and enhance user experience across all screen sizes.

**Screenshot**

Image

CC: @bodhish

### PR DESCRIPTION
**Pull request by modamaan**
## Proposed Changes

- Fixes #9369 
- Created Dropdown for Hidden Breadcrumb

Screenshot 2024-12-14 111923

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced a new breadcrumb navigation component with improved structure and semantics.
	- Added individual components for breadcrumb items, links, separators, and ellipsis for enhanced navigation experience.
	- Implemented a dropdown for additional breadcrumb items, enhancing interactivity.

- **Bug Fixes**
	- Improved handling of the "Back" button within the new breadcrumb structure.

- **Documentation**
	- Enhanced accessibility features in breadcrumb components for better user experience.

### CODE REVIEW

**Comment by modamaan** [comment]
@nihal467 can you review

File: src/components/Common/Breadcrumbs.tsx [hunk 1]
```diff
@@ -70,76 +82,86 @@ export default function Breadcrumbs({
...
-                </div>
+                <Button
+                  asChild
+                  variant="link"
+                  className="p-1 font-normal text-gray-800 underline underline-offset-2 hover:text-gray-700"
+                >
+                  <Link href="/">Home</Link>
+                </Button>
               </li>
+            </BreadcrumbItem>
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Common/Breadcrumbs.tsx]: Remove the li tags.

File: src/components/Common/Breadcrumbs.tsx [hunk 2]
```diff
@@ -70,76 +82,86 @@ export default function Breadcrumbs({
...
+                            •••
+                          </Button>
+                        </DropdownMenuTrigger>
+                        <DropdownMenuContent align="start">
+                          {crumbs.slice(0, -1).map((crumb, index) => (
+                            <DropdownMenuItem key={index}>
+                              <Button
+                                asChild
+                                variant="link"
+                                className="p-1 font-normal text-gray-800 underline underline-offset-2 hover:text-gray-700"
```
Comment (Jacobjeevan) [inline src/components/Common/Breadcrumbs.tsx:143]: Remove the underline styles.

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

**Comment by modamaan** [comment]
@Jacobjeevan can you please check the updation

**Comment by modamaan** [comment]
@rithviknishad

**Review by rithviknishad — APPROVED** [review]

**Review by Jacobjeevan — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Pull request merged**